### PR TITLE
add UI and logic forgot password

### PR DESCRIPTION
**Pull request by WalterGoldenOwl**
Description Pull Request:
## Todo
- Code UI and logic forgot password

## Task
Link of task: 
1. https://trello.com/c/jPak4ttE
2. https://trello.com/c/J4LRTIXh

## Screenshot
image

Screenshot_1696411438
Screenshot_1696411448

## Video

https://github.com/dustin-goldenowl/jobspot/assets/144195318/fda9f243-a8a2-4275-81fc-eeab4f170ff7

### CODE REVIEW

File: lib/src/presentations/forgot_password/screen/forgot_password_view.dart [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    return Scaffold(
+      body: BlocListener<ForgotPasswordCubit, ForgotPasswordState>(
+        listenWhen: (previous, current) {
+          if (previous.isLoading) Navigator.of(context).pop();
+          return true;
+        },
+        listener: (context, state) {
+          if (state.isLoading) loadingAnimation(context);
+
+          if (state.dataState is DataFailed) {
```
Comment (dustin-goldenowl) [inline lib/src/presentations/forgot_password/screen/forgot_password_view.dart:29]: If have defined `isSuccess`, `isError` in some PR before. You can use `state.dataState.isSucess` or `state.dataState.isError` in this case.

File: lib/src/presentations/forgot_password/screen/forgot_password_view.dart [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            customToast(context, text: state.dataState?.error ?? "");
+          }
+
+          if (state.dataState is DataSuccess) {
```
Comment (dustin-goldenowl) [inline lib/src/presentations/forgot_password/screen/forgot_password_view.dart:33]: the same above

File: lib/src/presentations/forgot_password/screen/forgot_password_view.dart (hunk 1 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/forgot_password/screen/forgot_password_view.dart:29]: In this case `state.dataState` can be `null` so it needs to be checked for `null` before calling it. The command will be longer!

File: lib/src/presentations/forgot_password/screen/forgot_password_view.dart (hunk 2 above)
Comment (WalterGoldenOwl) [reply lib/src/presentations/forgot_password/screen/forgot_password_view.dart:33]: In this case `state.dataState` can be `null` so it needs to be checked for `null` before calling it. The command will be longer!

**Pull request merged**